scenarios/pubsub_simple.erl

```diff
@@ -0,0 +1,288 @@
+        true ->
+            amoc_metrics:update_counter(node_create_success, 1),
+            amoc_metrics:update_time(node_create_time, CreateNodeTime);
+        Error ->
+            amoc_metrics:update(node_create_fails, 1),
+            lager:error("Error creating node: ~p", [Error]),
+            exit(connection_failed)
+    end,
+    Node.
+
+pubsub_node() ->
+    {?PUBSUB_ADDR, pubsub_node_name()}.
+
+rand_name(Prefix) ->
+    Suffix = base64:encode(crypto:strong_rand_bytes(5)),
+    <<Prefix/binary, "_", Suffix/binary>>.
+
+pubsub_node_name() ->
+    Name0 = rand_name(<<"princely_musings">>),
+    re:replace(Name0, "/", "_", [global, {return, binary}]).
+
+id(Client, {NodeAddr, NodeName}, Suffix) ->
+    UserName = escalus_utils:get_username(Client),
+    list_to_binary(io_lib:format("~s-~s-~s-~s",
+				 [UserName, NodeAddr, NodeName, Suffix])).
+
+subscribe(Client, Node) ->
+    Id = id(Client, Node, <<"subscribe">>),
+    Request = escalus_pubsub_stanza:subscribe(Client, Id, Node),
+    escalus:send(Client, Request),
+    {SubscribeTime, SubscribeResult} = timer:tc(
+        fun() ->
+            escalus:wait_for_stanza(Client, ?WAIT_FOR_STANZA_TIMEOUT)
+        end),
+    case escalus_pred:is_iq_result(SubscribeResult) of
+        true ->
+            amoc_metrics:update_counter(subscribe_success, 1),
+            amoc_metrics:update_time(subscribe_time, SubscribeTime);
+        Error ->
+            amoc_metrics:update_counter(subscribe_fails, 1),
+            lager:error("Error subscribing node ~p filed: ~p", [Node, Error]),
+            exit(connection_failed)
+        end.
+
+publish_pubsub_item(Client, Node) ->
+    Id = id(Client, Node, <<"publish">>),
+    ItemId = integer_to_binary(from_now(os:timestamp())),
+    Request = escalus_pubsub_stanza:publish(Client, ItemId, item_content(), Id, Node),
+    escalus:send(Client, Request),
+    {PublishTime, PublishResult} = timer:tc(
+        fun() ->
+            escalus:wait_for_stanza(Client, ?WAIT_FOR_STANZA_TIMEOUT)
+        end),
+    case escalus_pred:is_iq_result(PublishResult) of
+        true ->
+            amoc_metrics:update_counter(publish_success, 1),
+            amoc_metrics:update_time(publish_time, PublishTime);
+        Error ->
+            amoc_metrics:update_counter(publish_fails, 1),
+            lager:error("Error subscribing node ~p filed: ~p", [Node, Error]),
+            exit(connection_failed)
+        end.
+
+item_content() ->
+    #xmlel{name = <<"entry">>,
+           attrs = [{<<"xmlns">>, <<"http://www.w3.org/2005/Atom">>}]}.
+
+%% --- General Helpers ------------------------------------------------------------------------------------------------
+
+try_get_member([]) ->
+    Members = pg2:get_members(?GROUPNAME),
+    try_get_member(Members);
+try_get_member(Members) ->
+    L = length(Members),
+    N = erlang:phash2(self(), L) + 1,
+    lists:nth(N, Members).
+
+from_now({MegaSecs,Secs,Usecs}) ->
```

Comment:
There is no need for that functionality. You can:
1. Use `os:system_time(microsecond)` to get current timestamp in microsec, or:
2. Use https://github.com/esl/usec which was made exactly for this in times where there was no `os:system_time` API.